### PR TITLE
fix(core): ldml: remove 0-length arrays 🙀

### PR DESCRIPTION
**Pull request by srl295**
- use C99 flexible array member instead

#7098

@keymanapp-test-bot skip

### CODE REVIEW

File: core/include/ldml/keyboardprocessor_ldml.ts [hunk 1]
```diff
@@ -19,7 +19,10 @@
 // TODO-LDML: namespace com.keyman.core.ldml {
 /**
  * Constants for the KMXPlus data format
- * These are shared between the data access layer and the compiler
+ * These are shared between the data access layer and the compiler.
+ * Note that the section IDs (section_keys etc.) are 32 bit hex
+ * values that are designed to appear as text when written in little endian
+ * format, so 0x7379656b = 'keys'
```
Comment (mcdurdin) [inline core/include/ldml/keyboardprocessor_ldml.ts:25]: Would it be possible to do this translation dynamically in our typescript code so we can have nice strings here in the source .ts file?

And then the .h file could have e.g.

```
#define LDML_SECTIONID_KEYS 0x7379656B
#define LDML_SECTIONNAME_KEYS "keys"
```

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Everything LGTM except I am a little uncomfortable with the change to numeric identifiers in the headers -- I feel like that's hard to read and maintain. I'd like to see us dynamically generating those in the TS code for use in C++ (so, having a helper for them in the TS header).

File: core/include/ldml/keyboardprocessor_ldml.ts (hunk 1 above)
Comment (srl295) [reply core/include/ldml/keyboardprocessor_ldml.ts:25]: that makes sense as an approach.  I'll add a helper function in TS and use it in the generator.

**Comment by srl295** [comment]
- [ ] MSVC choking on the printf

File: core/include/ldml/keyboardprocessor_ldml.ts (hunk 1 above)
Comment (srl295) [reply core/include/ldml/keyboardprocessor_ldml.ts:25]: done

**Comment by srl295** [comment]
@mcdurdin 
> Cannot open include file: '../kmx/kmx_file.h': No such file or directory

merge conflict?

**Comment by mcdurdin** [comment]
> Cannot open include file: '../kmx/kmx_file.h': No such file or directory
> 
> merge conflict?

Which build is this from?

**Comment by srl295** [comment]
@mcdurdin  see if 16b8d725fd0527ac21c67c0f70d1837c7e8b0b7b fixes it

**Comment by srl295** [comment]
@mcdurdin ptal and go ahead and merge if this works

**Comment by mcdurdin** [comment]
Jenkins build fails with:

image

@ermshiperete any ideas?

**Comment by mcdurdin** [comment]
I'm going to go ahead and merge this and we'll solve the Jenkins build in a separate PR, because as it is this hiccup is block builds on all the other PRs in feature-ldml branch. (Isn't it always the way that we spend so much of our energy on silly build issues?)

**Review by mcdurdin — APPROVED** [review]

**Comment by mcdurdin** [comment]
This same compile error is happening everywhere on #7054, so we'll need to dig deeper.

**Pull request merged**